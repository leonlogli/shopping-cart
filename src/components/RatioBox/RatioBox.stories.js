import React from 'react'

import RatioBox from '.'

export default {
  title: 'RatioBox',
}

export const Default = () => (
  <RatioBox width={100} bgcolor="red" ratio={75}>
    Test
  </RatioBox>
)
