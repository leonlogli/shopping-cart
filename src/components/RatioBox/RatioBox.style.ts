import styled from 'styled-components'

import { Box } from '..'

const ContainerBox = styled(Box)`
  position: relative;
  width: 100%;

  &:before {
    content: '';
    display: block;
    padding-top: ${({ ratio }: any) => ratio}%; /* ratio of 1:0.75*/
  }
`

const InnerBox = styled(Box)`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  flex-direction: column;
`

export { ContainerBox, InnerBox }
