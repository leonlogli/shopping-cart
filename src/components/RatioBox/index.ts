export * from './RatioBox'
export { default } from './RatioBox'
