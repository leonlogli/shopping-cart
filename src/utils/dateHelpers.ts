const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
