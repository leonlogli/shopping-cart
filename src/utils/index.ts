export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
