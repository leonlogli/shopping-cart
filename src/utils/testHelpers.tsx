import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
