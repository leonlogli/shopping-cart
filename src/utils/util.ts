const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
