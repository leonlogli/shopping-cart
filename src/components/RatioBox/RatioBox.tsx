import React from 'react'

import { Box, BoxProps } from '../Layout'

import { ContainerBox, InnerBox } from './RatioBox.style'

export type RatioBoxProps = BoxProps & {
  /**
   * Ratio value in percent
   */
  ratio: number
}

const RatioBox = (props: RatioBoxProps) => {
  const { ratio, children, ...other } = props

  return (
    <Box {...other}>
      <ContainerBox {...{ ratio }}>
        <InnerBox>{children}</InnerBox>
      </ContainerBox>
    </Box>
  )
}

export { RatioBox }
export default RatioBox
