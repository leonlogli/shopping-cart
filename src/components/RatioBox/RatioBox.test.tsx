import React from 'react'

import { mountWithTheme } from '../../utils'

import RatioBox from './RatioBox'

describe('<RatioBox />', () => {
  const wrapper = mountWithTheme(<RatioBox ratio={75}>Test</RatioBox>)

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
