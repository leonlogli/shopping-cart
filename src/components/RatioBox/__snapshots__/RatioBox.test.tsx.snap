// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatioBox /> should match a snapshot 1`] = `
<RatioBox
  ratio={75}
>
  <ForwardRef>
    <Styled(Styled(MuiBox))>
      <Styled(MuiBox)
        className="hkFgNP"
      >
        <div
          className="MuiBox-root MuiBox-root-1 hkFgNP"
        >
          <Styled(Component)
            ratio={75}
          >
            <ForwardRef
              className="bZSHXs"
              ratio={75}
            >
              <Styled(Styled(MuiBox))
                className="bZSHXs"
                ratio={75}
              >
                <Styled(MuiBox)
                  className="hkFgNP bZSHXs"
                  ratio={75}
                >
                  <div
                    className="MuiBox-root MuiBox-root-2 hkFgNP bZSHXs"
                    ratio={75}
                  >
                    <Styled(Component)>
                      <ForwardRef
                        className="cmVGKu"
                      >
                        <Styled(Styled(MuiBox))
                          className="cmVGKu"
                        >
                          <Styled(MuiBox)
                            className="hkFgNP cmVGKu"
                          >
                            <div
                              className="MuiBox-root MuiBox-root-3 hkFgNP cmVGKu"
                            >
                              Test
                            </div>
                          </Styled(MuiBox)>
                        </Styled(Styled(MuiBox))>
                      </ForwardRef>
                    </Styled(Component)>
                  </div>
                </Styled(MuiBox)>
              </Styled(Styled(MuiBox))>
            </ForwardRef>
          </Styled(Component)>
        </div>
      </Styled(MuiBox)>
    </Styled(Styled(MuiBox))>
  </ForwardRef>
</RatioBox>
`;
